src/UI/Renderer.php

```diff
@@ -20,4 +22,19 @@
 	 * @return string
 	 */
 	public function render($component);
+
+	/**
+	 * Get a new renderer with an additional context.
+	 *
+	 * A context makes it possible to use another renderer for (some) components when
+	 * they are renderer as subcomponents of a certain components. The use case that
+	 * spawned this functionality is the observation, that e.g. items representing
+	 * repository objects are renderer in different lists, where the individual items
+	 * look different every time but are morally the same item. Another use case could
+	 * be a kiosk mode for replaying tests or using ILIAS a LTI tool provider.
+	 *
+	 * @param  Component	$context
+	 * @return Renderer
+	 */
+	public function withAdditionalContext(Component $context);
```

Comment:
For the use case you described I would rather switch the default renderer completely (if templates/css/less won't do). For the use cases we talked about before (items and listings, kiosk mode) I definitely see that there is a respective component wrapped around the stuff that is rendered. I somehow shied away from  plain strings, because typos, less clarity what that feature is good for, etc.